util/copy_file.go

```diff
@@ -0,0 +1,81 @@
+package util
```

Comment:
Would we want to move this to e.g. `internal/util/copy_file.go` to prevent others from importing and depending on these utilities?